docs/sc-memory/api/cpp/extended/agents/actions.md

```diff
@@ -0,0 +1,345 @@
+ScAddr const & argAddr = action.GetArgument(ScKeynodes::rrel_1);
+// If there is no argument with such role, then `argAddr` will be empty.
+...
+```
+
+```cpp
+...
+// Or provide index of argument. 1 is equal to `ScKeynodes::rrel_1`,
+// 2 is equal to `ScKeynodes::rrel_2` and etc.
+ScAddr const & argAddr = action.GetArgument(1);
+// If there is no argument with such role, then `argAddr` will be empty.
+...
+```
+
+!!! warning
+    This method will throw `utils::ExceptionInvalidParams` if you provide invalid index for argument (for example, 0).
+
+```cpp
+...
+ScAddr const & argAddr = action.GetArgument(1, defaultArgumentAddr);
+// If there is no argument with such role, then `argAddr` will equal 
+// to `defaultArgumentAddr`.
+...
+```
+
+### **GetArguments**
+
+You can get all arguments using one getters.
+
+```cpp
+...
+// Here, 2 is number arguments in specified action.
+auto const & [argAddr1, argAddr2] = action.GetArguments<2>();
+// If action doesn't have some argument, it will be empty.
+...
+```
+
+```cpp
+...
+// You can't ignore some arguments.
+auto const & [argAddr1, argAddr2, _] = action.GetArguments<3>();
+...
+```
+
+!!! note
+    `GetArguments` uses `GetArgument` for each argument.
+
+### **SetArgument**
+
+You can set arguments for specified action.
+
+```cpp
+...
+// Provide relation here if your argument should have specific role in action. 
+action.SetArgument(ScKeynodes::rrel_1, argAddr);
+...
+```
+
+```cpp
+...
+// Or provide index of argument. 1 is equal to `ScKeynodes::rrel_1`,
+// 2 is equal to `ScKeynodes::rrel_2` and etc.
+action.SetArgument(1, argAddr);
+...
+```
+
+!!! note
+    If action has already argument with specified role, then connection between action and this argument will be removed and created new one between action and new argument.
+
+### **SetArguments**
+
+```cpp
+...
+// Set several arguments simultaneously.
+action.SetArguments(argAddr1, argAddr2);
+...
+```
+
+!!! note
+    `SetArguments` uses `SetArgument` for each argument.
+
+### **Action result**
+
+All action should have result (result situation). Result situation is structure that contains all sc-constructions that which indicate result of interpreting action.
+
+#### **GetResult**
+
+You can result for any actions.
+
+```cpp
+...
+// Use this method after that some agent finished 
+// interpreting action.
+ScStructure const & actionResult = action.GetResult();
+...
+```
+
+!!! warning
+    You can call this method for finished action only.
+
+#### **SetResult**
+
+```cpp
+...
+// Use this method in agent interpreting action 
+// to set new result.
+action.SetResult(resultStructure);
+...
+```
+
+!!! note
+    If action has result, then will be removed and the new one will be set.
+
+!!! warning
+    You can call this method for not finished, but initiated action only.
+
+#### **FormResult**
+
+You must not create result structure. You can provide only elements of result for action.
```

Comment:
what if I want to set some edge as action result and not structure with this edge?

Reply:
Action result must be sc-structure

Reply:
https://ims.ostis.net/ is saying that nrel_result may be a conjunctive or a disjunctive expression as well

Reply:
It's not valid information. See Standard

Reply:
https://github.com/ostis-ai/ostis-standard/blob/main/Contents/part_ps/src/sd_agents.tex#L408 is also saying that result is not always an atomic logical formula

Reply:
Yes. But we're not going to support that now. If necessary, we will add it, but not yet

Reply:
ok, then change answer on similar question in faq section

Reply:
Okay

Reply:
Fixed